app/modules/auth/auth_router.py

```
@@ -52,52 +67,958 @@ async def login(login_request: LoginRequest):
+                        is_primary=False,  # Keep SSO as primary if it exists
+                    )
+                    unified_auth.add_provider(uid, provider_create)
+                    logger.info(f"Added firebase_email provider for user {uid}")
+
+                    return Response(
+                        content=json.dumps(
+                            {"uid": uid, "exists": True, "linked": True}
+                        ),
+                        status_code=200,
+                    )
+                except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)